heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -62,6 +74,13 @@ public Slave(SlaveLooper slaveLooper,
     this.streamInCommunicator = streamInCommunicator;
     this.streamOutCommunicator = streamOutCommunicator;
     this.inControlQueue = inControlQueue;
+    this.metricsOutCommunicator = metricsOutCommunicator;
+    this.serializer =
+        SerializeDeSerializeHelper.getSerializer(helper.getTopologyContext().getTopologyConfig());
+
+    // TODO (nlu): Create the state with corresponding type according to config
+    instanceState = new HashMapState<>();
```

Comment:
why have the Slave create empty state? shouldn't this be injected? there could be previous state to be loaded.